akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala

```diff
@@ -111,6 +118,62 @@ class AskSpec extends ScalaTestWithActorTestKit("""
       }
     }
 
+    "fail the future if the actor has terminated" in {
+      val actor: ActorRef[Msg] = spawn(behavior)
+
+      val deadLetterProbe = createDeadLetterProbe()
+      val probe = createTestProbe[Any]()
+      actor ! Stop(probe.ref)
+      implicit val timeout: Timeout = 10.millis
+
+      val answer: Future[String] = actor.ask(Foo("bar", _))
+      val result = answer.failed.futureValue
+      result shouldBe a[TimeoutException]
+      result.getMessage should startWith("Ask timed out on")
+
+      val deadLetter = deadLetterProbe.receiveMessage()
+      deadLetter.message match {
+        case Foo(s, _) => s should ===("bar")
+        case _         => fail(s"unexpected DeadLetter: $deadLetter")
+      }
+
+      val deadLettersRef = system.classicSystem.deadLetters
+      deadLetter.recipient shouldNot equal(deadLettersRef)
+      deadLetter.recipient should equal(ActorRefAdapter.toClassic(actor))
```

Comment:
Can't use `actor.toClassic`, There should be a better way to optimize this line.